lib/galaxy/model/__init__.py

```diff
@@ -1157,7 +1177,7 @@ def is_hda(d):
     return isinstance( d, HistoryDatasetAssociation )
 
 
-class History( object, Dictifiable, UsesAnnotations, HasName ):
+class History( HasTags, Dictifiable, UsesAnnotations, HasName ):
```

Comment:
I reverted back to the `HasTags` inherits from `object` way of doing things and then removed `object` from the inheritance list. Looks like the only way to tame the MRO beast.